src/libraries/Common/src/System/AppContextSwitchHelper.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+
+namespace System
+{
+    internal static class AppContextSwitchHelper
+    {
+        internal static bool GetBooleanConfig(string switchName, bool defaultValue = false) =>
+            AppContext.TryGetSwitch(switchName, out bool value) ? value : defaultValue;
+
+        internal static bool GetBooleanConfig(string switchName, string envVariable, bool defaultValue = false)
+        {
+            if (AppContext.TryGetSwitch(switchName, out bool value))
+            {
+                return value;
+            }
+
+            if (Environment.GetEnvironmentVariable(envVariable) is string str)
+            {
+                if (str == "1" || string.Equals(str, bool.TrueString, StringComparison.OrdinalIgnoreCase))
```

Comment:
Is this compared to '1' on purpose, for example in HTTP we do ` != 0`:
https://github.com/dotnet/runtime/blob/de1e6a431d20ffd110bc6c8ae39dd69176a2182b/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RuntimeSettingParser.cs#L30-L33

I see mix of both, I guess it doesn't matter that much...